File: java/2268. Minimum Number of Keypresses/Solution.java
class Solution {
    public int minimumKeypresses(String s) {
        Integer[] freq = new Integer[26];
        Arrays.fill(freq, 0);

        for (int i = 0; i < s.length(); i++) {
            freq[s.charAt(i) - 'a']++;
        }

        Arrays.sort(freq, Comparator.reverseOrder());

        int totalKeypress = 0;
        for (int i = 0; i < freq.length; i++) {
            totalKeypress += freq[i] * ((i + 9) / 9);
        }

        return totalKeypress;

    }

}

File: python/2268. Minimum Number of Keypresses/solution.py
class Solution:
    def minimumKeypresses(self, s: str) -> int:
        freq = [0] * 26
        totalKeyPress = 0

        for char in s:
            freq[ord(char) - ord('a')] += 1

        freq.sort(reverse=True)

        for i in range(len(freq)):
            totalKeyPress += freq[i] * ((i + 9) // 9)

        return totalKeyPress